src/main/java/org/htmlunit/javascript/host/URLSearchParams.java

```diff
@@ -137,26 +143,153 @@ public URLSearchParams() {
      */
     @JsxConstructor
     public URLSearchParams(final Object params) {
-        // TODO: Pass in a sequence
-        // new URLSearchParams([["foo", 1],["bar", 2]]);
-
-        // TODO: Pass in a record
-        // new URLSearchParams({"foo" : 1 , "bar" : 2});
-
         url_ = new URL("http://www.htmlunit.org", "");
 
         if (params == null || Undefined.isUndefined(params)) {
             return;
         }
 
         try {
-            url_.setSearch(splitQuery(Context.toString(params)));
+            url_.setSearch(splitQuery(createParamString(params)));
+        }
+        catch (final EcmaError e) {
+            throw ScriptRuntime.typeError("Failed to construct 'URLSearchParams': " + e.getErrorMessage());
         }
         catch (final MalformedURLException e) {
             LOG.error(e.getMessage(), e);
         }
     }
 
+    private static String createParamString(final Object params) {
+        // if params is a sequence
+        if (params instanceof Scriptable && ScriptableObject.hasProperty((Scriptable)params, SymbolKey.ITERATOR)) {
+            final Scriptable sequence = (Scriptable)params;
+            final StringBuilder sb = new StringBuilder();
+            final Iterator<Object> sequenceIterator = new IterationProtocolIterator(sequence, Context.getCurrentContext(), sequence.getParentScope());
```

Comment:
sb is kinda squashed.. how about for example this?
```suggestion
            final StringBuilder sb = new StringBuilder();

            final Iterator<Object> sequenceIterator = new IterationProtocolIterator((Scriptable)params,
                    Context.getCurrentContext(), ((Scriptable)params).getParentScope());
```